### PR TITLE
chore(tests): improve profiles test using gomock

### PR DESCRIPTION
**Pull request by dadamu**
## Description

Closes: #XXXX
This PR improves the test by applying gomock, which makes dependencies tests much easier.

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [x] targeted the correct branch (see [PR Targeting](https://github.com/desmos-labs/desmos/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://docs.cosmos.network/v0.44/building-modules/intro.html)
- [ ] included the necessary unit and integration [tests](https://github.com/desmos-labs/desmos/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: x/profiles/keeper/common_test.go [hunk 1]
```diff
@@ -137,55 +110,23 @@ func (suite *KeeperTestSuite) SetupTest() {
...
-	scopedProfilesKeeper := suite.capabilityKeeper.ScopeToModule(types.ModuleName)
-	suite.IBCKeeper = ibckeeper.NewKeeper(
-		suite.cdc,
-		keys[ibchost.StoreKey],
-		suite.paramsKeeper.Subspace(ibchost.ModuleName),
-		suite.stakingKeeper,
-		suite.upgradeKeeper,
-		scopedIBCKeeper,
-	)
+	// gomock initializations
```
Comment (RiccardoM) [inline x/profiles/keeper/common_test.go:113]:
```suggestion
	// Mocks initializations
```

File: x/profiles/keeper/genesis.go [hunk 2]
```diff
@@ -25,6 +25,20 @@ func (k Keeper) InitGenesis(ctx sdk.Context, data types.GenesisState) []abci.Val
...
 	k.SetParams(ctx, data.Params)
 
+	// Initialize the IBC settings
+	k.SetPort(ctx, data.IBCPortID)
+
+	// Only try to bind to port if it is not already bound, since we may already own
+	// port capability from capability InitGenesis
+	if !k.IsBound(ctx, data.IBCPortID) {
+		// module binds to the port on InitChain
+		// and claims the returned capability
```
Comment (RiccardoM) [inline x/profiles/keeper/genesis.go:35]:
```suggestion
		// Module binds to the port on InitChain and claims the returned capability
```

File: x/profiles/keeper/genesis_test.go [hunk 3]
```diff
@@ -183,14 +186,15 @@ func (suite *KeeperTestSuite) Test_InitGenesis() {
...
-		name      string
-		store     func(ctx sdk.Context)
-		genesis   *types.GenesisState
-		shouldErr bool
-		check     func(ctx sdk.Context)
+		name        string
+		setup       func(ctx sdk.Context)
+		store       func(ctx sdk.Context)
+		genesis     *types.GenesisState
+		shouldPanic bool
```
Comment (RiccardoM) [inline x/profiles/keeper/genesis_test.go:193]: Why was this renamed to `shouldPanic`? I would prefer leaving it as `shouldErr` instead to follow the same naming conventions as other tests

File: x/profiles/keeper/genesis_test.go [hunk 4]
```diff
@@ -183,14 +186,15 @@ func (suite *KeeperTestSuite) Test_InitGenesis() {
...
 
 	testCases := []struct {
-		name      string
-		store     func(ctx sdk.Context)
-		genesis   *types.GenesisState
-		shouldErr bool
-		check     func(ctx sdk.Context)
+		name        string
+		setup       func(ctx sdk.Context)
+		store       func(ctx sdk.Context)
```
Comment (RiccardoM) [inline x/profiles/keeper/genesis_test.go:191]: I'm actually quite puzzled on whether we should split `setup` and `store` into two different methods. Considering that in the `setup` we are actually telling how keepers should be answering, I think they can be merged into one single method (leaving the old `store` for backward compatibility and naming conventions). What do you think?

File: x/profiles/keeper/genesis_test.go (hunk 3 above)
Comment (dadamu) [reply x/profiles/keeper/genesis_test.go:193]: Since the failed result of the `InitGenesis` is a panic message, I think the behaviour is different from the error message so I though it was typos.

File: x/profiles/keeper/genesis_test.go (hunk 4 above)
Comment (dadamu) [reply x/profiles/keeper/genesis_test.go:191]: From my perspective, `store` should primarily focus on `ctx` statement, while `check` verify `ctx` statement after the operation is executed. As you mentioned, `setup` we are actually telling how keepers should be answering, these keepers are not involved with `ctx` statement, I believe it would be beneficial to separate them into two distinct methods.

**Review by manu0466 — APPROVED** [review]

File: x/profiles/keeper/genesis_test.go (hunk 4 above)
Comment (dadamu) [reply x/profiles/keeper/genesis_test.go]: Yes, removed it.

**Review by RiccardoM — APPROVED** [review]
Looks good to me
